utils/kubernetes_client.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+import requests
+import os
+
+requests.packages.urllib3.disable_warnings()
+
+from utils.conf import cfme_data, credentials
+
+"""
+Related yaml structures:
+
+[cfme_data]
+container_managers:
+    kubernetes:
+        name: My kubernetes
+        type: kubernetes
+        hostname: 10.12.13.14
+        port: 6443
+        credentials: kubernetes
+        authenticate: true
+        rest_protocol: https
+
+[credentials]
+kubernetes:
+    username: admin
+    password: secret
+    token: mytoken
+"""
+
+
+BASIC_ENTITIES = ['node', 'pod', 'service', 'replicationcontroller', 'namespace']
+NAMESPACE_DEPENDENT = ['pod', 'service', 'replicationcontroller']
+
+
+class KubernetesException(Exception):
+    pass
+
+
+class AuthException(KubernetesException):
+    pass
+
+
+class BearerTokenAuth(requests.auth.AuthBase):
+    """Attaches a bearer token to the given request object"""
+    def __init__(self, token):
+        self.token = token
+
+    def __call__(self, r):
+        r.headers['Authorization'] = 'Bearer {}'.format(self.token)
+        return r
+
+
+class KubernetesClient(object):
+
+    def __init__(self, hostname, auth, protocol="https", port=6443, entry='api/v1', verify=False):
+        self.api_entry = "{}://{}:{}/{}".format(protocol, hostname, port, entry)
+        self.verify = verify
+        if type(auth) in (list, set, tuple):
+            self.auth = auth
+        elif type(auth) is str:
+            self.auth = BearerTokenAuth(auth)
+        else:
+            AuthException('Invalid auth object')
```

Comment:
I know you would sooner or later find and fix this, but I cannot help myself ... `raise`